openedx/core/djangoapps/password_policy/compliance.py

```diff
@@ -0,0 +1,159 @@
+                deadline=strftime_localized(deadline, DEFAULT_SHORT_DATE_FORMAT)
+            )
+        )
+
+
+def _rollout_config():
+    """
+    Return a dictionary with configuration settings for managing the rollout of password policy compliance
+    enforcement.
+    """
+    return getattr(settings, 'PASSWORD_POLICY_COMPLIANCE_ROLLOUT_CONFIG', {})
+
+
+def _check_user_compliance(user, password):
+    """
+    Returns a boolean indicating whether or not the user is compliant with password policy rules.
+
+    A user is considered to be compliant if any of the following are true:
+    - They are a member of the PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME Group
+    - The provided password argument meets current password policy requirements.
+
+    In the event that a user that is not a member of the PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME Group is found to
+    be compliant, this method will attempt to add them to that Group before returning.
+    """
+
+    # Check if the user is a member of the compliant users group.
+    try:
+        if user.groups.filter(name=PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME).exists():
+            return True
+    except:
+        # If we fail to verify group membership, fallback to checking the password again.
+        pass
+
+    # Check if the user's password meets current policy requirements.
+    try
+        validate_password(password, user=user)
+        is_compliant = True
+    except:
+        # If anything goes wrong, we should assume the password is not compliant but we don't necessarily
+        # need to prevent login.
+        is_compliant = False
+
+    # If the password was confirmed to be compliant, add the user to compliant users group.
+    if is_compliant:
+        try:
+            user.groups.add(Group.objects.get(name=PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME))
+        except:
+            # Failing to add the user to the compliant users group is not a reason to prevent login.
+            pass
+
+    return is_compliant
+
+
+def _get_compliance_deadline_for_user(user):
+    """
+    Returns the date that the user will be required to comply with password policy rules, or None if no such date
+    applies to this user.
+    """
+    config = _rollout_config()
+
+    staff_deadline = config.get('STAFF_USER_COMPLIANCE_DEADLINE', None)
+    if staff_deadline:
+        staff_deadline = parse_date(staff_deadline)
+
+    elevated_privilege_user_deadline = config.get('ELEVATED_PRIVILEGE_USER_COMPLIANCE_DEADLINE', None)
+    if elevated_privilege_user_deadline:
+        elevated_privilege_user_deadline = parse_date(elevated_privilege_user_deadline)
+
+    general_user_deadline = config.get('GENERAL_USER_COMPLIANCE_DEADLINE', None)
+    if general_user_deadline:
+        general_user_deadline = parse_date(general_user_deadline)
+
+    if staff_deadline and user.is_staff:
+        return staff_deadline
+    elif elevated_privilege_user_deadline and user.courseaccessrole_set.exists():
```

Comment:
Per conversation with @bderusha , we should add error handling for the `courseaccessrole_set` call and use the  `general_user_deadline` when we encounter an error executing that query.